server/svix-server/src/error.rs

```diff
@@ -31,57 +28,71 @@ pub type Result<T> = std::result::Result<T, Error>;
 #[derive(Debug)]
 pub struct Error {
     // the file name and line number of the error. Used for debugging non Http errors
-    pub trace: Vec<&'static str>,
+    pub trace: Vec<&'static Location<'static>>,
     pub typ: ErrorType,
 }
 
 impl Error {
-    pub fn generic(s: impl fmt::Display, location: &'static str) -> Self {
+    #[track_caller]
+    pub fn generic(s: impl fmt::Display) -> Self {
```

Comment:
In general, I like this change. But there are a few places that have gotten less ergonomic, not more.

E.g.

```rust
.map_err(|e| err_generic!("Error parsing message timestamp {e:?}"))?;
```

becoming

```rust
.map_err(|e| Error::generic(format!("Error parsing message timestamp: {e:?}")))?;
```

Since `Error::generic` is now capturing the callsite automatically, what do you think about also changing these functions (`Error::generic`, `Error::database`, etc) to instead take `s: impl fmt::Debug`

And then
```rust
.map_err(|e| err_generic!("Error parsing message timestamp {e:?}"))?;
```

just becomes
```rust
.map_err(Err::generic)?;
```

Which is a helluva lot more ergonomic, and essentially doesn't lose any information (since we have the Location)